src/D2L.Bmx/Write.cs

```diff
@@ -0,0 +1,87 @@
+
+public class Write {
+	public static void WriteHandler( string? org, string? user, string? account, string? role, int? duration, bool nomask, string? output, string? profile ) {
+		// TODO: Get values of org, user, account, role and profile from the config file and assign them locally. This way we can check if
+		// values are null to see if they have been set through the commandline or config file
+
+		var unsetFlags = new List<string>();
+
+		// ask user to input org if org flag isn't set
+		if( string.IsNullOrEmpty( org ) ) {
+			unsetFlags.Add( "\"org\"" );
+		};
+
+		// check if profile flag has been set
+		if( string.IsNullOrEmpty( profile ) ) {
+			unsetFlags.Add( "\"profile\"" );
+		}
+
+		if( unsetFlags.Count != 0 ) {
+			Console.Write( "Error: required flag(s) " );
+			for( int i = 0; i < unsetFlags.Count; i++ ) {
+				Console.Write( unsetFlags.ElementAt( i ) );
+				if( i < unsetFlags.Count - 1 ) {
+					Console.Write( ", " );
+				} else {
+					Console.WriteLine( " not set" );
+				}
+			}
+			// Also print the output of the help command
+			return;
+		}
+
+		// ask user to input username if user flag isn't set
+		if( string.IsNullOrEmpty( user ) ) {
+			Console.Write( "Okta Username: " );
+			// there is currently no input validation for the username in the golang build, but it can be added
+			user = Console.ReadLine();
+		};
+
+		// Asks for user password input, or logs them in through caches
+		Authenticator.Authenticate( org, user, nomask );
+
+		// TODO: replace placeholder values with actual values
+		var accounts = new[] { "Dev-Slims", "Dev-Toolmon", "Int-Dev-NDE" };
+		var roles = new[] { "Dev-Slims-ReadOnly", "Dev-Slims-Admin" };
+
+		if( string.IsNullOrEmpty( account ) ) {
```

Comment:
the account & role selection code in the print & the write commands is identical, right?
may be suitable to refactor into its own function.